### ISSUE TITLE
[linter] fix ui::mod::calc_scroll_top, missing_const_for_fn warning

### ISSUE THREAD

**Issue by Horki** [extrawurst/gitui#301]
**Describe the bug, warning**
```
warning: this could be a `const fn`
  --> src/ui/mod.rs:10:1
   |
10 | / pub fn calc_scroll_top(
11 | |     current_top: usize,
12 | |     height_in_lines: usize,
13 | |     selection: usize,
...  |
21 | |     }
22 | | }
   | |_^
   |
note: the lint level is defined here
  --> src/main.rs:10:9
   |
10 | #![warn(clippy::missing_const_for_fn)]
```

**To Reproduce**
`cargo +stable clippy --release`

**Context (please complete the following information):**
 - OS/Distro + Version: [e.g. ` 5.8.12-arch1-1 `]
 - GitUI Version [e.g. `0.10.1`]
 - Rust version: [e.g `1.44`]

@extrawurst PR #300

**Comment by extrawurst** [comment]
I intentionally made those *warnings* in https://github.com/extrawurst/gitui/blob/master/src/main.rs#L10 to not break the build, this is only until clippy stops complaining without noticing I cannot make it const due to `saturating_sub`.

the reason I want to keep it a warning is to not forget about it

**Comment by Horki** [comment]
@extrawurst reference #300 , "fixed in rust 1.47"; I'll close this
